homeassistant/components/axis/device.py

```diff
@@ -227,6 +263,7 @@ async def get_device(hass, host, port, username, password):
         with async_timeout.timeout(15):
 
             await asyncio.gather(
```

Comment:
It's not good to await many executor jobs concurrently. We risk exhausting the pool. Just await them in a loop.

Reply:
Or better, make a function that calls all the sync functions and just schedule this wrapper function on the executor. We'll have less context switches that way.

Reply:
Will fix!

Reply:
Depending on device these requests can be quite heavy/slow. Thats why Ive run them separately. You still prefer one function?

Reply:
How long time are we talking?

Reply:
Some seconds. I don't remember exactly

Reply:
I will also add more calls later on so it will increase. The new apis are quicker but the old ones are quite large and slow

Reply:
Seconds are fine. Minutes is too much.